### PR TITLE
Email verification link

### PR DESCRIPTION
**Pull request by Aierie**

### CODE REVIEW

File: packages/hub/node-tests/tasks/send-email-card-drop-verification-test.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    params.append('verification-code', cardDropRequest.verificationCode);
+    params.append('email-hash', cardDropRequest.emailHash);
+    let link = 'https://card-drop-email.test/email-card-drop/verify' + '?' + params.toString();
+
+    await task.perform({ email: 'anyone@test.test', id: cardDropRequest.id }, helpers);
+
+    expect(StubEmail.lastSent.to).to.equal('anyone@test.test');
+    expect(StubEmail.lastSent.from).to.equal(config.get('aws.ses.supportEmail'));
+    expect(StubEmail.lastSent.text).to.contain(link);
+    expect(StubEmail.lastSent.html).to.contain(link);
```
Comment (Aierie) [inline packages/hub/node-tests/tasks/send-email-card-drop-verification-test.ts:96]: I'm waiting for the template we're meant to use to be finalised before asserting on that, so leaving this a bit relaxed for now.

File: packages/hub/node-tests/tasks/send-email-card-drop-verification-test.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    );
+
+    expect(StubEmail.lastSent).to.equal(null);
+
+    await waitFor(() => testkit.reports().length > 0);
+
+    let sentryReport = testkit.reports()[0];
+    expect(sentryReport.tags).to.deep.equal({
+      event: 'send-email-card-drop-verification',
+    });
```
Comment (backspace) [inline packages/hub/node-tests/tasks/send-email-card-drop-verification-test.ts:132]: Quite unrelated to this but is there some way we can extract this Sentry-checking, lines like this are now scattered everywhere 😯

File: packages/hub/tasks/send-email-card-drop-verification.ts [hunk 3]
```diff
@@ -1,7 +1,67 @@
...
+    params.append('eoa', request.ownerAddress);
+    params.append('verification-code', request.verificationCode);
+    params.append('email-hash', request.emailHash);
+    const verificationLink = config.get('cardDrop.verificationUrl') + '?' + params.toString();
+
+    const senderEmailAddress = config.get('aws.ses.supportEmail') as string;
+
+    const emailTitle = 'Claim your Card Drop';
+    const emailBodyHtml = `<h1></h1> This is your verification link: <a href="${verificationLink}">${verificationLink}</a>`;
+    const emailBodyText = `This is your verification link: ${verificationLink}`;
```
Comment (backspace) [inline packages/hub/tasks/send-email-card-drop-verification.ts:44]: Seeing this potential site of XSS (is it? or injection of some kind), is there any possibility that someone can submit a fake EOA?

**Review by backspace — APPROVED** [review]

File: packages/hub/node-tests/tasks/send-email-card-drop-verification-test.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    updated_at: createdAt,
+    locked_at: null,
+    locked_by: null,
+    revision: 0,
+    key: null,
+    flags: null,
+  };
+}
+
+let helpers: Helpers = makeJobHelpers({}, makeMockJob('remove-old-sent-notifications'), {
```
[the remark this answers is not in the document]
Comment (Aierie) [reply packages/hub/node-tests/tasks/send-email-card-drop-verification-test.ts:38]: Oops. This is a copy paste error

File: packages/hub/tasks/send-email-card-drop-verification.ts (hunk 3 above)
Comment (Aierie) [reply packages/hub/tasks/send-email-card-drop-verification.ts:44]: It does seem possible and we should probably sanitize this. Unclear whether SES does this for us, so will create a linear issue for this.

File: packages/hub/node-tests/tasks/send-email-card-drop-verification-test.ts (hunk 2 above)
Comment (Aierie) [reply packages/hub/node-tests/tasks/send-email-card-drop-verification-test.ts:132]: Unsure, it is pretty common. Probably can make a utility function or custom chai assertion?

Comment (Aierie) [reply packages/hub/node-tests/tasks/send-email-card-drop-verification-test.ts:132]: https://www.chaijs.com/guide/helpers/#adding-methods

**Pull request merged**